pandas/core/arrays/categorical.py

```diff
@@ -1135,14 +1134,17 @@ def remove_categories(self, removals):
         [NaN, 'c', 'b', 'c', NaN]
         Categories (2, object): ['b', 'c']
         """
+        from pandas import Index
+
         if not is_list_like(removals):
             removals = [removals]
 
-        removals = {x for x in set(removals) if notna(x)}
+        removals = Index(removals).dropna().unique()
```

Comment:
Wouldn't the other way round be faster? e.g. unique().dropna()?

Reply:
Good point. Its not measurable with that benchmark, but I swapped it anyway.